### PR TITLE
dbmysql.class.php - nextOOP() & phpdoc

### PR DESCRIPTION
**Pull request by AnaelMobilia**
dbmysql.class.php
  - Add function nextOOP() (discussed on #1143 )
  - Add / update phpdoc for the whole file

### CODE REVIEW

File: inc/dbmysql.class.php [hunk 1]
```diff
@@ -243,16 +222,13 @@ function prepare($query) {
...
    /**
-    * Give result from a mysql result
-    *
-    * @param $result    MySQL result handler
-    * @param $i         Row to give
-    * @param $field     Field to give
-    *
-    * @return Value of the Row $i and the Field $field of the Mysql $result
-   **/
+    * Give result from a sql result
```
Comment (trasher) [inline inc/dbmysql.class.php]: Most of the time, blocks into doblocks are separated with a blank line (description -> blank line -> parameters -> blank line -> return).
I don't know if this is something required with générators, but I thinks it is more easy to read :)

**Comment by trasher** [comment]
I'm not fan on the method name :p

I'd prefer something like `nextObject()`; rename `next()` to `nextAssoc()` and bind `next()` to `nextAssoc()`.

What do you think?

**Comment by AnaelMobilia** [comment]
Why not :-)
I will do it !

File: inc/dbmysql.class.php (hunk 1 above)
Comment (AnaelMobilia) [inline inc/dbmysql.class.php]: +1 
https://phpdoc.org/docs/latest/getting-started/your-first-set-of-documentation.html#example

Maybe we should create a § on readthedocs ?

Comment (trasher) [inline inc/dbmysql.class.php]: Not a bad idea :)

File: inc/dbmysql.class.php [hunk 2]
```diff
@@ -57,42 +52,33 @@ class DBmysql {
...
 
 
    /**
     * Constructor / Connect to the MySQL Database
     *
-    * try to connect
-    *
-    * @param $choice integer, host number (default NULL)
-    *
-    * @return nothing
```
Comment (trasher) [inline inc/dbmysql.class.php]: Not sure the @return is not mandatory. With the cs rule I've proposed; this is required to have a `@return` tag even on void methods.

Comment (AnaelMobilia) [inline inc/dbmysql.class.php]: The [phpDocumentor](https://phpdoc.org/docs/latest/references/phpdoc/tags/return.html#description) explain :
```
It is RECOMMENDED when documenting to use this tag with every function and method. Exceptions to this recommendation are:
  1. constructors, the @return tag MAY be omitted here, in which case @return self is implied.
  2. functions and methods without a `return` value, the @return tag MAY be omitted here, in which case @return void is implied.
```

I think we are on the 1. case ?

Comment (trasher) [inline inc/dbmysql.class.php]: Yes, I agree this is not mandatory. But I do not know if that can be excluded from CS doc checks :/

Comment (AnaelMobilia) [inline inc/dbmysql.class.php]: I added @return on every function :)

Comment (trasher) [inline inc/dbmysql.class.php]: Great, thank you!! :)

**Comment by trasher** [comment]
I've rebased to fix UT fails; and I've also suqashed ;)

**Comment by trasher** [comment]
@AnaelMobilia could you try with the latest file and tell me if your editor is still happy? Then I'll merge this one :)

**Comment by AnaelMobilia** [comment]
@trasher Thanks for the following !
It's all good for me :)

**Pull request merged**